app/components/base_component.rb

```diff
@@ -8,7 +8,11 @@ def before_render
   end
 
   def self.scripts
-    @scripts ||= _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+    @scripts ||= begin
+      scripts = _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+      scripts.concat superclass.scripts if superclass.respond_to?(:scripts)
```

Comment:
I think one of the things that was bugging me about the previous implementation was the specific `BaseComponent` being hardcoded, but switching this to `superclass.respond_to?(:scripts)` is way better IMO